ovs/datapath.go

```diff
@@ -0,0 +1,342 @@
+type ConnTrackReader interface {
+	// GetCTLimits is the method used to querying conntrack limits for a
+	// datapath on a switch
+	GetCTLimits(string, []uint64) (ConnTrackOutput, error)
+}
+
+// ConnTrackWriter is the interface defining the write operations
+// of ovs conntrack
+type ConnTrackWriter interface {
+	// SetCTLimits is the method used to setup a limit for an ofport (zone)
+	// belonging to a datapath of  a switch
+	SetCTLimits(string) (string, error)
+	// DelCTLimits is the method used to remove a limit to an ofport (zone)
+	// belonging to a datapath of a switch
+	DelCTLimits(string, []uint64) (string, error)
+}
+
+// CLI is an interface defining a contract for executing a command.
+// Implementation of shell cli is done by the Client concrete type
+type CLI interface {
+	Exec(args ...string) ([]byte, error)
+}
+
+// DataPathService defines the concrete type used for DataPath operations
+// supported by the ovs-dpctl command
+type DataPathService struct {
+	// We define here a CLI interface making easier to mock ovs-dpctl command
+	// as in github.com/digitalocean/go-openvswitch/ovs/datapath_test.go
+	CLI
+}
+
+// NewDataPathService is a builder for the DataPathService.
+// sudo is defined as a default option.
+func NewDataPathService() *DataPathService {
+	return &DataPathService{
+		CLI: &OvsCLI{
+			c: New(Sudo()),
+		},
+	}
+}
+
+// Version retruns the ovs-dptcl --version currently installed
+func (dp *DataPathService) Version() (string, error) {
+	args := []string{"--version"}
+	result, err := dp.CLI.Exec(args...)
+	if err != nil {
+		return "", err
+	}
+
+	return string(result), nil
+}
+
+// GetDataPaths returns the output of the command 'ovs-dpctl dump-dps'
+func (dp *DataPathService) GetDataPaths() ([]string, error) {
+	args := []string{"dump-dps"}
+	result, err := dp.CLI.Exec(args...)
+	if err != nil {
+		return nil, err
+	}
+
+	return strings.Split(string(result), "\n"), nil
+}
+
+// AddDataPath create a Datapath with the command 'ovs-dpctl add-dp <DP>'
+// It takes one argument, the required DataPath Name and returns an error
+// if it failed
+func (dp *DataPathService) AddDataPath(dpName string) ([]byte, error) {
+	args := []string{"add-dp", dpName}
+
+	return dp.CLI.Exec(args...)
+}
+
+// DelDataPath create a Datapath with the command 'ovs-dpctl del-dp <DP>'
+// It takes one argument, the required DataPath Name and returns an error
+// if it failed
+func (dp *DataPathService) DelDataPath(dpName string) ([]byte, error) {
+	args := []string{"del-dp", dpName}
+
+	return dp.CLI.Exec(args...)
+}
+
+// GetCTLimits returns the conntrack limits  for a given datapath
+// equivalent to running: 'sudo ovs-dpctl ct-get-limits <datapath_name> zone=<#1>,<#2>,...'
+func (dp *DataPathService) GetCTLimits(dpName string, zones []uint64) (*ConnTrackOutput, error) {
+	// Start by building the args
+	args := []string{"ct-get-limits"}
+	if dpName == "" {
+		return nil, errMissingDataPathName
+	}
+
+	args = append(args, dpName)
```

Comment:
nit: could simplify by combining L157 above with this line like this:

```
args := []string{"ct-get-limits", dpName}
```

In this case, since the arguments are variable, using a slice and passing it as variadic argument seems appropriate.